solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,668 @@
+  type        = string
+  description = "The expiry time for root CA Certificate Revocation List (CRL)"
+  default     = null
+  validation {
+    condition     = var.root_ca_crl_expiry == null ? true : can(regex("^[0-9]+[s,m,h,d]{0,1}$", var.root_ca_crl_expiry))
+    error_message = "root_ca_crl_expiry must match regular expression /^[0-9]+[s,m,h,d]{0,1}$/"
+  }
+}
+
+variable "root_ca_crl_disable" {
+  type        = bool
+  description = "crl_disable flag for the root CA"
+  default     = false
+}
+
+variable "root_ca_crl_distribution_points_encoded" {
+  type        = bool
+  description = "crl_distribution_points_encoded flag for the root CA"
+  default     = true
+}
+
+variable "root_ca_issuing_certificates_urls_encoded" {
+  type        = bool
+  description = "issuing_certificates_urls_encoded flag for the root CA"
+  default     = true
+}
+
+##############################################################################
+# Intermediate CA
+##############################################################################
+
+variable "intermediate_ca_name" {
+  type        = string
+  description = "Name of the Intermediate CA to create for a private_cert secret engine"
+  default     = "intermediate-ca"
+}
+
+variable "intermediate_ca_max_ttl" {
+  type        = string
+  description = " for the intermediate CA"
+  default     = "26300h"
+}
+
+variable "intermediate_ca_common_name" {
+  type        = string
+  description = "Common name for the intermediate CA"
+  default     = "cloud.ibm.com"
+}
+
+variable "intermediate_ca_crl_expiry" {
+  type        = string
+  description = "crl_expiry for the intermediate CA"
+  default     = "72h"
+}
+
+variable "intermediate_ca_crl_disable" {
+  type        = bool
+  description = "crl_disable for the intermediate CA"
+  default     = false
+}
+
+variable "intermediate_ca_crl_distribution_points_encoded" {
+  type        = bool
+  description = "crl_distribution_points_encoded flag for the intermediate CA"
+  default     = true
+}
+
+variable "intermediate_ca_issuing_certificates_urls_encoded" {
+  type        = bool
+  description = "issuing_certificates_urls_encoded flag for the intermediate CA"
+  default     = true
+}
+
+variable "intermediate_ca_signing_method" {
+  type        = string
+  description = "The signing method to use with this certificate authority to generate private certificates"
+  default     = "internal"
+
+  validation {
+    condition     = contains(["internal", "external"], var.intermediate_ca_signing_method)
+    error_message = "specified signing_method is not valid. Allowed values are: internal, external"
+  }
+}
+
+##############################################################################
+# Template
+##############################################################################
+
+variable "certificate_template_name" {
+  type        = string
+  description = "Name of the Certificate Template to create for a private_cert secret engine"
```

Comment:
the description needs to be updated to mention about the naming convention used when using prefix